SecurityPkg/Library/AuthVariableLib/AuthService.c

```diff
@@ -605,10 +718,13 @@ ProcessVarWithPk (
   // Init state of Del. State may change due to secure check
   //
   Del = FALSE;
-  if (  (InCustomMode () && UserPhysicalPresent ())
+  // MU_CHANGE [START] SecurityPkg/Secureboot: Support Fix delete PK failing bug
+  if (  (!IsVariablePolicyEnabled ())

```

Comment:
Would this change the behavior? I thought PP is still needed when the system is enlightened to do so?

Reply:
So alternatively I could do the following:

```c
  if (  ((InCustomMode () && UserPhysicalPresent ()) || (!IsVariablePolicyEnabled ()))
   || (  (mPlatformMode == SETUP_MODE)
        && !(FeaturePcdGet (PcdRequireSelfSignedPk) && IsPk)))
  {
 ```
 
 Which would keep the existing behavior. The reason I removed InCustomMode(..) and UserPhysicallyPresent(..) is because we take a stance in Project MU that we won't enable CUSTOM MODE nor will we use UserPhysicallyPresent(..)
 
So InCustomMode(..) will always be FALSE, and even if the system is UserPhysicallyPresent(..) enlightend it won't matter because CUSTOM MODE will always be false (for Project MU).

Reply:
I agree you could consider that condition as well to not deviate from the upstream (e.g. (!IsVariablePolicyEnabled() || (InCustomMode() && UserPhysicalPresent())) || ...).

Reply:
I wrote that while you were also responding. I think this could go different ways depending on how the overall project wants to treat consideration of custom and physical presence. My understanding was that those APIs are in place similar to the upstream (not deprecated / removed in a way a platform wouldn't be able to use them) but specified negatively as a Mu platform decision.

If a wider stance is taken that will never be used, it might make sense to ASSERT if present, so platforms don't have to guess where they're being taken into account versus the upstream.

Reply:
Yeah I think your initial understanding is still true. I don't think we can affirmatively say that a downstream platform wont make that decision it's just that we don't recomend it.

There are other examples where similiar patterns are used such as ProcessVariable(..)

```c
  if ((!EFI_ERROR (Status)) && IsDeleteAuthVariable (OrgVariableInfo.Attributes, Data, DataSize, Attributes) && (UserPhysicalPresent () || !IsVariablePolicyEnabled ())) {

``` 

We might need a larger discussion / more changes if we want to take that wider stance